Config/CMakeIncludes/FindTasmanianMAGMA.cmake

```diff
@@ -0,0 +1,248 @@
+#        FATAL_ERROR is raised if missing all versions of magma or _sparse
+#
+#    if USE_XSDK_DEFAULTS is OFF
+#        use cmake find_library()
+#        HINTS will be generated from the value of Tasmanian_MAGMA_ROOT_DIR
+#
+#    both shared and static libraries are considered,
+#    if all variants of either magma or magma_sparse end on ".a" suffix,
+#        then Tasmanian_MAGMA_SHARED_FOUND is set OFF
+#
+#    search for the include folder considers in order
+#        ${Tasmanian_MAGMA_ROOT_DIR}/include
+#        ${Tasmanian_MAGMA_ROOT_DIR}
+#        <path>/../../include
+#        <path>/../include
+#        <path>/include
+#    where <path> is the folder containing libmagma.so (or .a)
+#
+#    NOTE: if Tasmanian_MAGMA_ROOT_DIR is missing,
+#              find_library() will still be called
+#              this should work if CMAKE_PREFIX_PATH is set correctly
+#
+# NOTE: the complexity of the logic is needed to ensure
+#       - both static and shared libs are found (if available)
+#       - the user can explicitly specify libraries (if desired)
+#       - compliance with xSDK policies (if using xSDK defaults)
+#
+#    Additional confusion can be created by projects that compile static
+#    libraries with position independent code, effectively creating
+#    shared libraries with .a extension. Thus, the check for the shared
+#    library extension cannot be considered as FATAL_ERROR
+
+macro(Tasmanian_magma_searchlib)
+    # usage: Tasmanian_magma_searchlib(ARGV0, ARGV1, ARGV2, ARGV3)
+    # ARGV0 -> VAR in the call to find_library(), returns the result
+    # ARGV1 -> NAMES in the call to find_lbrary(), e.g., magma or magma_sparse
+    # ARGV2 -> ON/OFF, ON adds NO_DEFAULT_PATH to the search
+    # ARGV3 -> (optional) path to be used for HINTS in find_library()
+    # HITS is searched before system paths
+    # and if Tasmanian_MAGMA_ROOT is set, it should be searched before system paths
+    if (${ARGC} GREATER 3)
+        if (${ARGV2}) # xSDK mode is ON, do no search default paths
+            find_library(${ARGV0} ${ARGV1} HINTS "${ARGV3}"
+                                           HINTS "${ARGV3}/lib/"
+                                           HINTS "${ARGV3}/lib/${CMAKE_CXX_LIBRARY_ARCHITECTURE}/"
+                                           HINTS "${ARGV3}/${CMAKE_CXX_LIBRARY_ARCHITECTURE}/lib/")
+        else()
+            message(STATUS "Searchign without defaults")
+            find_library(${ARGV0} ${ARGV1} HINTS "${ARGV3}"
+                                           HINTS "${ARGV3}/lib/"
+                                           HINTS "${ARGV3}/lib/${CMAKE_CXX_LIBRARY_ARCHITECTURE}/"
+                                           HINTS "${ARGV3}/${CMAKE_CXX_LIBRARY_ARCHITECTURE}/lib/"
+                                           NO_DEFAULT_PATH NO_CMAKE_PATH)
+        endif()
+    else()
+        find_library(${ARGV0} ${ARGV1}) # don't have any hints, hope CMAKE_PREFIX_PATH is set
+    endif()
```

Comment:
Anyway IMO, too complicated.  Keep it simple...

Reply:
I think cmake fails when I try to pass "" to the macro and then reference it. I will double-check.